mdsuite/file_io/lammps_trajectory_files.py

```diff
@@ -57,222 +62,247 @@
 
-        return int(number_of_lines / (number_of_atoms + self.header_lines))
+        with open(self.file_path, "r") as file:
+            file.seek(0)
+            for _ in tqdm.tqdm(range(n_batches)):
+                yield self._read_process_n_configurations(file, self._batch_size)
+            if n_configs_remainder > 0:
+                yield self._read_process_n_configurations(file, n_configs_remainder)
 
-    def _get_time_information(self, number_of_atoms: int):
+    def _read_process_n_configurations(
+        self, file, n_configs
+    ) -> mdsuite.file_io.file_read.TrajectoryChunkData:
         """
-        Get time information.
-
+        Read n_configs configurations and bring them to the structore needed for the yield of get_configurations_generator()
         Parameters
         ----------
-        number_of_atoms : int
-                Number of atoms in each trajectory
+        file
+            The open trajectory file. Note: Calling this function will advance the reading point in the file
+        n_configs
 
         Returns
         -------
 
         """
-        header = self._read_header(self.f_object)  # get the first header
-        time_0 = float(header[1][0])  # Time in first configuration
-        header = self._read_header(
-            self.f_object, offset=number_of_atoms
-        )  # get the second header
-        time_1 = float(header[1][0])  # Time in second configuration
-        self.f_object.seek(0)  # return to first line in file
-
-        return time_1 - time_0
+        chunk = mdsuite.file_io.file_read.TrajectoryChunkData(
+            self._mdata.species_list, n_configs
+        )
 
-    def _get_species_information(self, number_of_atoms: int):
+        for config_idx in range(n_configs):
+            # skip the header
+            mdsuite.file_io.file_read.skip_n_lines(file, self.n_header_lines)
+            # read one config
+            traj_data = np.stack(
+                [
+                    np.array(list(file.readline().split()))
+                    for _ in range(self._n_particles)
+                ]
+            )
+            # sort by id
+            if not self.trajectory_is_sorted_by_ids:
+                traj_data = sort_array_by_column(traj_data, self._id_column_idx)
+
+            # slice by species
+            for sp_info in self._mdata.species_list:
+                idxs = self._species_dict[sp_info.name]["line_idxs"]
+                sp_data = traj_data[idxs, :]
+                # slice by property
+                for prop_info in sp_info.properties:
+                    prop_column_idxs = self._property_dict[prop_info.name]
+                    write_data = sp_data[:, prop_column_idxs]
+                    # add 'time' axis. we only have one configuration to write
+                    write_data = write_data[np.newaxis, :, :]
+                    chunk.add_data(write_data, config_idx, sp_info.name, prop_info.name)
```

Comment:
this also seems to be a common method, which differes mostly by where/if to add a `np.newaxis`